control/db.py

```diff
@@ -94,16 +94,16 @@ class OmapPersistentConfig(PersistentConfig):
     HOST_PREFIX = "host_"
     LISTENER_PREFIX = "listener_"
 
-    def __init__(self, nvme_config):
+    def __init__(self, settings):
```

Comment:
It looks like this class includes 3 different concepts that IMO should be separated: local config, omap config, update
My suggestion is to change the name of the base class `PersistentConfig` to just `Config` and have two sub-classes: `LocalConfig` and `OmapConfig`. Add a function to retrieve the current version to the base. Add another sub-class to `Config` that e.g. is called `ConsistentConfig` where the constructor takes two configs as arguments where in this example one would be local the other omap. All calls like e.g. add_bdev would funnel through the `ConsistentConfig` class which updates both underlying configs and makes sure they are aligned, i.e. have the same version. The whole update code can also be put in this `ConsistentConfig` class. To compute the diff you can add another function that allows reading all keys from either.